src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/ApplicationService.kt

```diff
@@ -63,4 +63,49 @@ class ApplicationService(
 
     return ValidatableActionResult.Success(createdApplication)
   }
+
+  fun updateApplication(applicationId: UUID, data: String, submittedAt: OffsetDateTime?, username: String): AuthorisableActionResult<ValidatableActionResult<ApplicationEntity>> {
+    val application = applicationRepository.findByIdOrNull(applicationId)
+      ?: return AuthorisableActionResult.NotFound()
+
+    val probationOfficer = probationOfficerService.getProbationOfficerForRequestUser()
+
+    if (application.createdByProbationOfficer != probationOfficer) {
+      return AuthorisableActionResult.Unauthorised()
+    }
+
+    if (application.submittedAt != null) {
+      return AuthorisableActionResult.Success(
+        ValidatableActionResult.GeneralValidationError("This application has already been submitted")
+      )
+    }
+
+    val validationErrors = mutableMapOf<String, String>()
```

Comment:
We're doing this a bit now, wondering if it might be worth having a `ValidationErrors` class? Just idle thoughts at this point, so may not be worth doing!